applications/IGAStructuralMechanicsApplication/custom_elements/base_discrete_element.h

```diff
@@ -0,0 +1,449 @@
+    * @param rDampingMatrix The elemental damping matrix
+    * @param rCurrentProcessInfo The current process info instance
+    */
+    void CalculateDampingMatrix(
+        MatrixType& rDampingMatrix,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /**
+    * @brief Sets on rResult the ID's of the element degrees of freedom
+    * @param rResult The vector containing the equation id
+    * @param rCurrentProcessInfo The current process info instance
+    */
+    void EquationIdVector(
+        EquationIdVectorType& rResult,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /**
+    * @brief Sets on rElementalDofList the degrees of freedom of the considered element geometry
+    * @param rElementalDofList The vector containing the dof of the element
+    * @param rCurrentProcessInfo The current process info instance
+    */
+    void GetDofList(
+        DofsVectorType& rElementalDofList,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /**
+    * @brief This function provides a more general interface to the element.
+    * @details It is designed so that rLHSvariables and rRHSvariables are passed to the element thus telling what is the desired output
+    * @param rLeftHandSideMatrix container with the output Left Hand Side matrix
+    * @param rRightHandSideVector container for the desired RHS output
+    * @param rCurrentProcessInfo the current process info instance
+    */
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /**
+    * @brief This is called during the assembling process in order to calculate the elemental right hand side vector only
+    * @param rRightHandSideVector the elemental right hand side vector
+    * @param rCurrentProcessInfo the current process info instance
+    */
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /**
+    * @brief Sets on rValues the nodal displacements
+    * @param rValues The values of displacements
+    * @param Step The step to be computed
+    */
+    void GetValuesVector(
+        Vector& rValues,
+        int Step = 0
+    ) override;
+
+    /**
+    * @brief Sets on rValues the nodal velocities
+    * @param rValues The values of velocities
+    * @param Step The step to be computed
+    */
+    void GetFirstDerivativesVector(
+        Vector& rValues,
+        int Step = 0
+    ) override;
+
+    /**
+    * @brief Sets on rValues the nodal accelerations
+    * @param rValues The values of accelerations
+    * @param Step The step to be computed
+    */
+    void GetSecondDerivativesVector(
+        Vector& rValues,
+        int Step = 0
+    ) override;
+
+
+/********************************************************************/
+/*    CalculateOnIntegrationPoints                                  */
```

Comment:
Why do you override all these functions even though they do nothing?
=> seems like a duplication of `element.h`
Plus keep in mind that this changes in #2194 !